### PR TITLE
LLM: add initial QuickStart for linux cpp usage

### PR DESCRIPTION
**Pull request by rnwang04**
### 1. Why the change?

We need to provide a QuickStart to show how to use llama.cpp with BigDL-LLM.

### 2. User API changes

No.

### 3. Summary of the change 

add initial QuickStart for linux cpp usage, the usage is worked with https://github.com/intel-analytics/llm.cpp/pull/278

preview: https://ruonantestdoc.readthedocs.io/en/cpp_quickstart/doc/LLM/Quickstart/llama_cpp_quickstart.html

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

**Comment by Oscilloscope98** [comment]
You may also need to update the TOC related link in these three places:

https://github.com/intel-analytics/BigDL/blob/main/docs/readthedocs/source/_templates/sidebar_quicklinks.html#L20
https://github.com/intel-analytics/BigDL/blob/main/docs/readthedocs/source/_toc.yml#L43
https://github.com/intel-analytics/BigDL/blob/main/docs/readthedocs/source/doc/LLM/Quickstart/index.rst

**Comment by rnwang04** [comment]
Have updated to latest usage, https://ruonantestdoc.readthedocs.io/en/cpp_quickstart/doc/LLM/Quickstart/llama_cpp_quickstart.html, would you mind taking a look ? @jason-dai

File: docs/readthedocs/source/doc/LLM/Quickstart/llama_cpp_quickstart.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
+# Use llama.cpp with BigDL-LLM on Intel GPU
+
+Now you can use BigDL-LLM as an Intel GPU accelerated frontend of [llama.cpp](https://github.com/ggerganov/llama.cpp). This quickstart guide walks you through setting up and using [llama.cpp](https://github.com/ggerganov/llama.cpp) with `bigdl-llm` on Intel GPU (both iGPU and dGPU).
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Quickstart/llama_cpp_quickstart.md:3]: backend

Comment (rnwang04) [reply docs/readthedocs/source/doc/LLM/Quickstart/llama_cpp_quickstart.md]: Have updated to `backend`.

**Comment by rnwang04** [comment]
Hi @jason-dai , do you think we can merge this PR as an initial version?   we can add windows support in next PR.

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**